rebar.config

```diff
@@ -29,6 +29,7 @@
 {plugins, [ rebar3_proper
           , coveralls
           , {rebar3_lint, {git, "https://github.com/project-fifo/rebar3_lint.git", {tag, "0.1.11"}}}
+          , {rebar3_erlang_ls, {git, "git@github.com:erlang-ls/rebar3_erlang_ls.git", {branch, "master"}}}
```

Comment:
Why is this plugin necessary? What benefits does it bring?

Reply:
The plugin is there so that we can use Erlang LS with Erlang LS itself (since it's a rebar3 project itself).

The purpose of the plugin is to run an agent (a `gen_server`) which has the rebar3 state (paths and the alike) in its own state, so that we can RPC commands to it, effectively using the Erlang API of `rebar3`.

Let's look at how, for example, running CT tests would work for a `rebar3` project without such a plugin.

One option would be to simply start a new Erlang node and run something like `ct:run_test/1` (basically what we had so far, but with a RPC in the mix). This would work for very simple cases, but it would fail as soon as the users would start setting `ct_opts` in their rebar.config or rely on hooks and the alike. We would have to re-implement half of the rebar3 CT provider for things to work correctly.

Another option would be to just `os:cmd` the `rebar3` commands, using the CLI interface rather than the Erlang one. But how would we get back the failing stacktrace in case of errors? Things are just outputted to stdout without our hook (it's not possible to pass custom hooks via the CLI interface of rebar3). We would have to play with group leaders and the alike.

A third option would be to use the `rebar3 shell` as the _runtime node_. The problem there is that the shell also starts apps and running CT tests results in the shell itself being killed.

So, the plugin is a compromise. It's annoying that we would require a rebar3 user to install it to work with the language server, but I guess that's still better than re-implementing half of rebar3 in Erlang LS. For projects which already have the concept of a _running node_, things would just work as long as you have a MFA which can reply via RPC. But I'm still not sure how it would work for a non-rebar3 project which does not have the concept of an _agent_. Open to suggestions, though.

Reply:
Maybe we should not default to the rebar3 implementation and keep the basic CT RPC on the server side as a default.

Reply:
I will proceed and merge for now. We can eventually follow-up.